src/main/resources/templates/Answer.html

```diff
@@ -0,0 +1,25 @@
+<!DOCTYPE html SYSTEM "http://www.thymeleaf.org/dtd/xhtml1-strict-thymeleaf-spring4-4.dtd">
+<html lang="en"
+      xmlns="http://www.w3.org/1999/xhtml"
+      xmlns:th="http://www.thymeleaf.org">
+<head>
+  <title>Getting Started: Serving Web Content</title>
+  <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
+</head>
+<body>
+
+<h1 th:text="${surveyName}"></h1>
+
+<form action="#" th:action="@{'/survey/'+${surveyID}+'/answer'}" th:object="${response}" method="post">
```

Comment:
seems like you're posting to /survey/{id}/answer, but I don't see anything that corresponds to this
answers (for now seem to) go to /surveys/{surveyID}/question/{questionID}/responses

Reply:
potential fix: 1 question per page, post for that question and go to next question rather than somehow posting for multiple questions with just html

Reply:
Forgot to add surveyRepository.save(s); 
I think that should fix the issue